osu.Game/Graphics/Containers/SectionsContainer.cs

```diff
@@ -150,31 +158,63 @@ public override void Add(T drawable)
             footerHeight = null;
         }
 
+        private ScheduledDelegate? scrollToTargetDelegate;
+
         public void ScrollTo(Drawable target)
         {
-            lastKnownScroll = null;
+            Logger.Log($"Scrolling to {target}..");
 
-            // implementation similar to ScrollIntoView but a bit more nuanced.
-            float top = scrollContainer.GetChildPosInContent(target);
+            lastKnownScroll = null;
 
-            float bottomScrollExtent = scrollContainer.ScrollableExtent;
-            float scrollTarget = top - scrollContainer.DisplayableContent * scroll_y_centre;
+            float scrollTarget = getScrollTargetForDrawable(target);
 
-            if (scrollTarget > bottomScrollExtent)
+            if (scrollTarget > scrollContainer.ScrollableExtent)
                 scrollContainer.ScrollToEnd();
             else
                 scrollContainer.ScrollTo(scrollTarget);
 
             if (target is T section)
                 lastClickedSection = section;
+
+            // Content may load in as a scroll occurs, changing the scroll target we need to aim for.
+            // This scheduled operation ensures that we keep trying until actually arriving at the target.
+            scrollToTargetDelegate?.Cancel();
+            scrollToTargetDelegate = Scheduler.AddDelayed(() =>
+            {
+                if (scrollContainer.UserScrolling)
+                {
+                    Logger.Log("Scroll operation interrupted by user scroll");
+                    scrollToTargetDelegate?.Cancel();
+                    scrollToTargetDelegate = null;
+                    return;
+                }
+
+                if (Precision.AlmostEquals(scrollContainer.Current, scrollTarget, 1))
+                {
+                    Logger.Log($"Finished scrolling to {target}!");
+                    scrollToTargetDelegate?.Cancel();
+                    scrollToTargetDelegate = null;
+                    return;
+                }
+
+                if (!Precision.AlmostEquals(getScrollTargetForDrawable(target), scrollTarget, 1))
+                {
+                    Logger.Log($"Reattempting scroll to {target} due to change in position");
+                    ScrollTo(target);
+                }
+            }, 50, true);
```

Comment:
The `50` here is arbitrary. Alternative would be running every frame in `UpdateAfterChildren` similar to the other section selection code.